### PR TITLE
Add tracking feature for Client and API

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
-  Task: https://github.com/idmc-labs/helix2.0-meta/issues/784#issue-1735487160

## Depends on:
- Server branch: https://github.com/idmc-labs/helix-server/pull/492#issue-1718820772

## Changes

- Add feature to keep track of the client records as well as API records
- Add feature to add/edit client in the client records

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

**Review by shreeyash07 — APPROVED** [review]

**Comment by tnagorra** [comment]
- [ ] Add date range filter
- [ ] Filter should be applied on export

File: src/components/forms/ClientRecordForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+        onPristineSet,
+        onValueSet,
+    } = useForm(defaultFormValues, schema);
+
+    const {
+        notify,
+        notifyGQLError,
+    } = useContext(NotificationContext);
+
+    const communicationVariables = useMemo(
```
Comment (tnagorra) [inline src/components/forms/ClientRecordForm/index.tsx:157]:
```suggestion
    const clientVariables = useMemo(
```

File: src/components/forms/ClientRecordForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+                }
+                const { errors, result } = createClientRes;
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors));
+                    notifyGQLError(errors);
+                    onErrorSet(formError);
+                }
+                if (onClientCreate && result) {
+                    notify({
+                        children: 'New Client record created successfully!',
```
Comment (tnagorra) [inline src/components/forms/ClientRecordForm/index.tsx:209]:
```suggestion
                        children: 'Client created successfully!',
```

File: src/components/forms/ClientRecordForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                }
+                const { errors, result } = updateClientRes;
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors));
+                    notifyGQLError(errors);
+                    onErrorSet(formError);
+                }
+                if (onClientCreate && result) {
+                    notify({
+                        children: 'Client record updated successfully!',
```
Comment (tnagorra) [inline src/components/forms/ClientRecordForm/index.tsx:249]:
```suggestion
                        children: 'Client updated successfully!',
```

File: src/views/ApiUsage/ApiRecordsTable/ApiRecordsFilters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        apiType: __type(name: "ExternalApiType") {
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
+
+// eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (tnagorra) [inline src/views/ApiUsage/ApiRecordsTable/ApiRecordsFilters/index.tsx:41]: Why do we need this?

File: src/views/ApiUsage/ApiRecordsTable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            apiQueryFilters,
+        ],
+    );
+
+    const totalApiCount = ApiData?.clientTrackInformationList?.totalCount ?? 0;
+    const apiRecords = ApiData?.clientTrackInformationList?.results;
+
+    const columns = useMemo(
+        () => ([
+            // FIXME: sort by api_type
```
Comment (tnagorra) [inline src/views/ApiUsage/ApiRecordsTable/index.tsx:238]: Let's remove this FIXME

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**